contracts/gateway/src/contract/execute.rs

```diff
@@ -102,3 +72,49 @@ pub fn route_outgoing_messages(
             .map(|msg| GatewayEvent::MessageRouted { msg }.into()),
     ))
 }
+
+fn partition_by_verified(
+    deps: DepsMut,
+    msgs: Vec<NewMessage>,
+) -> error_stack::Result<(Vec<NewMessage>, Vec<NewMessage>), ContractError> {
+    let verifier = load_config(&deps)?.verifier;
+
+    let query_msg = aggregate_verifier::msg::QueryMsg::IsVerified {
+        messages: msgs.clone(),
+    };
+    let query_response: Vec<(CrossChainId, bool)> = deps
+        .querier
+        .query(&QueryRequest::Wasm(WasmQuery::Smart {
+            contract_addr: verifier.to_string(),
+            msg: to_binary(&query_msg).map_err(ContractError::from)?,
+        }))
+        .map_err(ContractError::from)?;
+
+    Ok(msgs.into_iter().partition(|msg| -> bool {
+        match query_response.iter().find(|r| msg.cc_id == r.0) {
+            Some((_, v)) => *v,
+            None => false,
+        }
+    }))
+}
+
+fn load_config(deps: &DepsMut) -> error_stack::Result<Config, ContractError> {
+    let cfg = CONFIG
+        .load(deps.storage)
+        .change_context(ContractError::ConfigNotFound)?;
```

Comment:
Shouldn't this be a panic? When would a config not exist?

Reply:
I guess we could panic here. I don't feel strongly about it. It's not dependent on user input, but the error gives a nicer response if it ever should happen. I'll think about it some more

Reply:
yes you are right, this makes more sense as a panic. I'll change it in a follow up PR that touches this code anyway